### PR TITLE
 Adding new weather provider for weather.gov

### PR DESCRIPTION
**Pull request by vincep5**
I've added a new weather provider for the "new" weather module.  Its an open API from weather.gov a US only provider of weather information.  It might be useful for those that do not want to signup for an account for the other weather providers.

### CODE REVIEW

File: modules/default/weather/providers/weathergov.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	fetchCurrentWeather() {
+		this.fetchData(this.getUrl())
+			.then(data => {
+				if (!data || !data.properties || !data.properties.periods || !data.properties.periods.length) {
+					// Did not receive usable new data.
+					// Maybe this needs a better check?
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.name}, ${data.sys.country}`);
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: Please remove commented code.

File: modules/default/weather/providers/weathergov.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	fetchWeatherForecast() {
+		this.fetchData(this.getUrl())
+			.then(data => {
+				if (!data || !data.properties || !data.properties.periods || !data.properties.periods.length) {
+					// Did not receive usable new data.
+					// Maybe this needs a better check?
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.city.name}, ${data.city.country}`);
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: Please remove commented code.

File: modules/default/weather/providers/weathergov.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+		return this.config.apiBase + this.config.lat + "," + this.config.lon + this.config.weatherEndpoint;
+	},
+
+	/*
+	 * Generate a WeatherObject based on currentWeatherInformation
+	 */
+	generateWeatherObjectFromCurrentWeather(currentWeatherData) {
+		const currentWeather = new WeatherObject(this.config.units);
+
+		//currentWeather.humidity = currentWeatherData.main.humidity;
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: Please remove commented code.

File: modules/default/weather/providers/weathergov.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+		currentWeather.temperature = currentWeatherData.temperature;
+		currentWeather.windSpeed = currentWeatherData.windSpeed.split(" ", 1);
+		currentWeather.windDirection = this.convertDirectiontoDegrees(currentWeatherData.windDirection);
+		currentWeather.weatherType = this.convertWeatherType(currentWeatherData.shortForecast, currentWeatherData.isDaytime);
+		//currentWeather.sunrise = moment(currentWeatherData.sys.sunrise, "X");
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: Please remove commented code.

File: modules/default/weather/providers/weathergov.js [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+		} else if (weatherType.includes("Smoke")) {
+			return "smoke";
+		} else if (weatherType.includes("Haze")) {
+			return "day-haze";
+		}
+
+		return null;
+	},
+
+	/* getParams(compliments)
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: getParams has no parameter called `compliments`, please remove

File: modules/default/weather/providers/weathergov.js [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+	/*
+	 * Convert the icons to a more usable name.
+	 */
+	convertWeatherType(weatherType, isDaytime) {
+		//https://w1.weather.gov/xml/current_obs/weather.php
+		// There are way too many types to create, so lets just look for certain strings
+
+		if (weatherType.includes("Cloudy") || weatherType.includes("Partly")) {
+			if (isDaytime) {
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: In this function you can save some else statements:

```js
if (isDaytime) {
  return "day-cloudy";
}

return "night-cloudy";
```

File: modules/default/weather/providers/weathergov.js [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+	},
+
+	/*
+	 * fetch forecast information for daily forecast.
+	 */
+	fetchForecastDaily(forecasts) {
+		// initial variable declaration
+		const days = [];
+		// variables for temperature range and rain
+		var minTemp = [];
```
Comment (fewieden) [inline modules/default/weather/providers/weathergov.js]: please do not mix `var` with `const` and `let` in this function. Stick to `const` and `let` like in the rest of the file.

**Comment by vincep5** [comment]
@fewieden thanks for the review. I have sent up another commit to fix those points you mentioned.  I also fixed openweathermap.js so that const, let, and var declarations are not mixed together.  I removed the var from both files.

**Comment by fewieden** [comment]
@vincep5 looks cleaner now

**Pull request merged**